core/vote/vote_manager.go

```diff
@@ -164,6 +169,21 @@ func (voteManager *VoteManager) loop() {
 				voteManager.pool.PutVote(voteMessage)
 				votesManagerCounter.Inc(1)
 			}
+		case event := <-voteManager.syncVoteCh:
+			voteMessage := event.Vote
+			if voteManager.eth.IsMining() || !voteManager.signer.UsingKey(&voteMessage.VoteAddress) {
```

Comment:
Will it panic if `voteManager.signer` is nil?

Reply:
voteManager.signer is assigned when New VoteManager
if voteManager.signer is nil，then `NewVoteManager` will fail, so loop will not to start